ext/dcos-installer/dcos_installer/action_lib/__init__.py

```diff
@@ -151,35 +163,37 @@ def _remove_host(state_file, host):
 
 
 @asyncio.coroutine
-def install_dcos(config, block=False, state_json_dir=None, hosts=[], async_delegate=None, try_remove_stale_dcos=False,
-                 roles=None, **kwargs):
-    if roles is None:
-        roles = ['master', 'agent']
+def install_dcos(config, block=False, state_json_dir=None, hosts=None, async_delegate=None, try_remove_stale_dcos=False,
+                 **kwargs):
+    if hosts is None:
```

Comment:
Can we add an assert that everything in the hosts list is an instance of Nodes if hosts is given? (This code will make really odd error messages if that isn't the case)